Issue/12 UI part to report (#13)
teamulster2/report: report/03_methodology/02_architecture.md

## Architecture

In this part of the report, the architecture of the application is described in more detail.

### Architecture overview

> Overall, the application is divided into two completely independent part. 

Reviewer: ```suggestion
Overall, the application is divided into two completely independent parts. 
```

> On the one hand, there is the stand-alone client app, which contains the self-test and a questionnaire. 

Reviewer: ```suggestion
On one hand, there is the stand-alone client app, which contains the self-test and a questionnaire. 
```

> On the other hand there is the server application that gives an export functionality with the user data.

Reviewer: ```suggestion
On the other hand, there is the server application that gives an export functionality for the collected research data.
```

#### Stand-alone client application

The client app includes the self-test, which is supposed to give reliable data about the test person several times a day after it has been carried out. 
The *Psychomotor vigilance task* and *Spatial span task* are two tasks included in the self-test. 

> There is also a questionnaire, which is supposed to record the current condition of the subject at longer intervals. 

Reviewer: ```suggestion
There is also a questionnaire which is supposed to record the current condition of the subject at longer intervals. 
```

This happens via a certain number of questions with four choices each. 

> There should also be a notification for the self-test, which asks the user up to three times a day to also carry out the test. 

Reviewer: ```suggestion
There should also be a notification for the self-test, which encourages the user up to three times a day to carry out the tests. 
```

> There should also be a notification for the self-test, which encourages the user up to three times a day to carry out the tests. 

Reviewer: ```suggestion
Notifications encourage the user up to three times per day to open the app and take the self-test.
```

> In addition, the current mood and current activity of the test person is saved before each self-test so that the data can also be reliably interpreted. 

Reviewer: ```suggestion
In addition, the current mood and current activity of the test person is saved before each self-test so that the data can also be interpreted reliably. 
```

> In addition, the current mood and current activity of the test person is saved before each self-test so that the data can also be interpreted reliably. 

Reviewer: ```suggestion
In addition to that, the current mood and current activity of the test person is saved before each self-test so that the data can also be interpreted reliably. 
```

> Last but not least, the client application also offers a settings page with options for client-server communication and to control this in a targeted manner.

Reviewer: ```suggestion
Last but not least, the client application offers a settings page with options for client-server communication and to control this in a targeted manner.
```

As the client application is written in *dart* with the *Flutter* framework you can create multiple cross platform apps out of a single codebase [6]. 

> This means that the client side is not limited to a single platform or operating system and the code does not have to be extended for other end devices.

Reviewer: ```suggestion
This results in the client side not being limited to a single platform or operating system and the source code not having to be extended for other devices.
```

#### Server application

// TODO

### User, client and server communication

For the communication between the user, the client an the server, one can say that the user only communicates with the client app. 

> The client app then forwards the data to the server at a suitable time.

Reviewer: ```suggestion
The client app then forwards the data to the server if the user agrees to share them.
```

> The first time the app is used it gets a config file from the server, which initially load the information for the user. 

Reviewer: ```suggestion
The first time the app is used the user needs to enter the study server url so the app can fetch a config file from the server. This file initializes the app with all necessary information. 
```

> This architecture has the advantage that the user does not necessarily have to rely on a stable internet connection when carrying out the self-test or the questionnaire, and the data can also be transmitted later. 

Reviewer: ```suggestion
This architecture has the advantage that the user does not necessarily have to rely on a stable internet connection when carrying out the self-test or the questionnaire, and the data can also be transmitted after a later session. 
```

> This architecture has the advantage that the user does not necessarily have to rely on a stable internet connection when carrying out the self-test or the questionnaire, and the data can also be transmitted after a later session. 

Reviewer: ```suggestion
Architecture designs like this offer the advantage that the user does not necessarily have to rely on a stable internet connection when carrying out the self-test or the questionnaire, and the data can also be transmitted after a later session. 
```

At the same time, the low level of data exchange also ensures that the user only receives relevant information to save on his own device. 

> In addition, the user is always aked whether the data should really be transmitted to the server. 

Reviewer: ```suggestion
In addition, the user is always asked whether the data should really be transmitted to the server. 
```

> In figure 1 you can see the communication between user, client and server in detail.

Reviewer: Maybe add a link to figure 1 🤔

Author: Sounds like a really good idey, do you know how to do that without markdown and pandoc? The only solutions I found used Latex, so I did not focus on that.

Reviewer: Maybe you could something like this `[![image alt text](image url link)](anchor_link)` and refer to the anchor link from within the text as normal link like `[test](anchor_link)`

> In addition, the user is always asked whether the data should really be transmitted to the server. 

Reviewer: ```suggestion
Additionally, after every completed self-test, the user is asked whether or not the data should be transmitted to the server. 
```

## Architecture

In this part of the report, the architecture of the application is described
in more detail.

### Architecture overview

Overall, the application is divided into two completely independent parts. On
one hand, there is the stand-alone client app, which contains the self-test
and a questionnaire. On the other hand, there is the server application that
gives an export functionality for the collected research data.

#### Stand-alone client application

The client app includes the self-test, which is supposed to give reliable data
about the test person several times a day after it has been carried out. The
*Psychomotor vigilance task* and *Spatial span task* are two tasks included in
the self-test. There is also a questionnaire which is supposed to record the
current condition of the subject at longer intervals. This happens via a
certain number of questions with four choices each. Notifications encourage

> the user up to three times per day to open the app and take the self-test. In

Reviewer 2: ```suggestion
the user to open the app and take the self-test. The Default value for the notification Interval is three times a day. In
```
